src/hotspot/share/classfile/defaultMethods.cpp

```diff
@@ -1091,7 +1078,7 @@ static void merge_in_new_methods(InstanceKlass* klass,
   int orig_idx = 0;
   int new_idx = 0;
 
-  for (int i = 0; i < new_size; ++i) {
+  for (u2 i = 0; i < new_size; ++i) {
```

Comment:
Is this change for the loop index necessary? Even as an int it is constrained to be <= max(u2) - and `at_put` expects an int index doesn't it?

Reply:
okay